### PR TITLE
Rework NowPlaying codec info in overlay

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Solves an issue reported in [this forum post](https://forum.kodi.tv/showthread.php?tid=359717&pid=3138761#pid3138761).

Issue was caused by Kodi server introducing new aspect ratios and a mis-design in loading the aspect ratio images in the remote app. The new aspect ratio "2.00" resulted in loading the image for "2" (channels).

I now decided to rework this in a way which does not break if new aspect ratios or number of audio channels are introduced.  Now the numbers shown are `UILabel` text fields. This allows to write e.g. aspect ratios like "2.34" (rightmost screenshot). In addition, I also changed the font size and type for better readability of bitrate / sampling rate / bits per sample, especially for bigger screens.

While testing I came across few potential problems which could cause wrong or incomplete A/V codec data to be displayed. This is fixed by using the correct `currentPlayerID` (the current playing player ID, not the playlist ID select by user) when checking which content to display. Also the loading of overlay data and codec data is changed. A/V codec data is read via "XBMC.GetInfoLabel" and this changes asynchronously. Therefore A/V codec data must be read repeatedly and cannot be read only once after the played item changes. An example is the aspect ratio which might take a while to settle after a change of the played video.

Screenshots: https://abload.de/img/bildschirmfoto2023-02z9fr9.png

## Summary for release notes

Bugfix: Ensure correct A/V codec data in NowPlaying overlay
Improvement: Support new supported aspect ratios
Improvement: Better readability of NowPlaying detail popover

### CODE REVIEW

**Comment by wutschel** [comment]
@kambala-decapitator, more changes than I wanted to, but this is worth being added to 1.12.1

File: XBMC Remote/NowPlaying.m [hunk 1]
```diff
@@ -133,6 +134,60 @@ - (NSString*)processSongCodecName:(NSString*)codec {
...
+    }
+    else if ([channels isEqualToString:@"8"]) {
+        channels = @"7.1";
+    }
+    else if ([channels isEqualToString:@"9"]) {
+        channels = @"8.1";
+    }
+    else if ([channels isEqualToString:@"10"]) {
+        channels = @"9.1";
+    }
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:170]: I think using a static dictionary would be better

File: XBMC Remote/NowPlaying.m [hunk 2]
```diff
@@ -133,6 +134,60 @@ - (NSString*)processSongCodecName:(NSString*)codec {
...
+    else if ([aspect isEqualToString:@"1.33"]) {
+        aspect = @"4:3";
+    }
+    else if ([aspect isEqualToString:@"1.78"]) {
+        aspect = @"16:9";
+    }
+    else if ([aspect isEqualToString:@"2.00"]) {
+        aspect = @"2:0";
+    }
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:187]: also dictionary

File: XBMC Remote/NowPlaying.m [hunk 3]
```diff
@@ -133,6 +134,60 @@ - (NSString*)processSongCodecName:(NSString*)codec {
...
+        aspect = @"1:1";
+    }
+    else if ([aspect isEqualToString:@"1.33"]) {
+        aspect = @"4:3";
+    }
+    else if ([aspect isEqualToString:@"1.78"]) {
+        aspect = @"16:9";
+    }
+    else if ([aspect isEqualToString:@"2.00"]) {
+        aspect = @"2:0";
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:186]: 2:0? what is that? :) 2:1 probably?

File: XBMC Remote/NowPlaying.m [hunk 4]
```diff
@@ -485,8 +544,20 @@ - (void)getActivePlayers {
...
+                        // Pause the A/V codec updates until Kodi's info labels settled
+                        waitForInfoLabelsToSettle = YES;
+                        [self performSelector:@selector(setWaitForInfoLabelsToSettle) withObject:nil afterDelay:1.0];
                     }
                 }
+                // Switch off overlay if the picture player is active
+                if (currentPlayerID == PLAYERID_PICTURES) {
+                    [self toggleSongDetails];
+                }
+                // Codec View uses "XBMC.GetInfoLabels" which might change asynchronously. Therefor check each time
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:556]:
```suggestion
                // Codec View uses "XBMC.GetInfoLabels" which might change asynchronously. Therefore check each time
```

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: XBMC Remote/NowPlaying.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: True, looks a lot better now.

File: XBMC Remote/NowPlaying.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: True, looks a lot better now.

File: XBMC Remote/NowPlaying.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: :) Infinite wide screen format

Fixed.

File: XBMC Remote/NowPlaying.m (hunk 4 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: Fixed and already squashed.

**Review by kambala-decapitator — APPROVED** [review]

**Comment by wutschel** [comment]
Squashed and rebase to master.

**Pull request merged**